core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -140,7 +140,8 @@ $(document).ready(function() {
             (attempt.hasOwnProperty("attemptId") ? attempt["attemptId"] + "/" : "") + "logs";
```

Comment:
Here the existence of the `attemptId` is checked but a few line below you are using it without checking its existence. When it is missing then the new link won't be valid too.

Reply:
The `attemptId` is not checked with line of `<td><a href="{{uiroot}}/history/{{id}}/{{attemptId}}/jobs/">{{attemptId}}</a></td>` in `historypage-template.html`. And need to ignore this link when `attempId` is empty or null. Do you have some ideas?

Reply:
The `{{uiroot}}/history/{{id}}/{{attemptId}}/jobs/">{{attemptId}}</a></td>` is protected by the check `{{#hasMultipleAttempts}} ... {{/hasMultipleAttempts}}`.

I think if there is only one attempt then the `attemptId` might be missing and the value 1 was used for that before your change not null and empty (because of the line `var num = app["attempts"].length;`).   

So please test with that case too. 
And I would like to ask you to attach some data to the PR: during testing you can put a breakpoint (ie by placing `debugger;` into the code) before the Mustache renders the page (it is about line 161) please copy-paste the `data` content to a code block of the PR comment.
 
Could you please do that for both having multiple attempts and having only one?

Reply:
+1 for @attilapiros 's comments.

Reply:
> The `{{uiroot}}/history/{{id}}/{{attemptId}}/jobs/">{{attemptId}}</a></td>` is protected by the check `{{#hasMultipleAttempts}} ... {{/hasMultipleAttempts}}`.
> 
> I think if there is only one attempt then the `attemptId` might be missing and the value 1 was used for that before your change not null and empty (because of the line `var num = app["attempts"].length;`).
> 
> So please test with that case too.
> And I would like to ask you to attach some data to the PR: during testing you can put a breakpoint (ie by placing `debugger;` into the code) before the Mustache renders the page (it is about line 161) please copy-paste the `data` content to a code block of the PR comment.
> 
> Could you please do that for both having multiple attempts and having only one?

I think this change is ok. The debugger data as follows:

This pic is when `hasMultipleAttempts` is true and `showCompletedColumns` is false.
![reliao_img_1579763662512](https://user-images.githubusercontent.com/8609142/72964152-0a291300-3df4-11ea-8eae-713849341e23.png)


Next pic is when `hasMultipleAttempts` is false and `showCompletedColumns` is true. There is only one attempt then the attemptId also exist. But to prevent attemptID missing, I will set attemptID to 1 before it.
![reliao_img_1579764031165](https://user-images.githubusercontent.com/8609142/72964171-1dd47980-3df4-11ea-9fcd-99d39234e2e3.png)